### PR TITLE
tss: Replacing sysfs based tpm detection with automatic lookup through get caps 

### PR DESCRIPTION
**Pull request by ChriMarMe**
This just probes the /dev/tpm0 path, depending on return of GetCapability function determine the tpm version and returns the TPM struct.

Signed-off-by: Christopher Meis <christopher.meis@9elements.com>

### CODE REVIEW

**Comment by ChriMarMe** [comment]
This can be testes if a PR in go-tpm is merged and the dependency is updated

File: pkg/tss/tss.go [hunk 1]
```diff
@@ -12,28 +12,49 @@ import (
...
+	t, err := tpm1.OpenTPM(tpmPath)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, tpm := range candidateTPMs {
-		tss, err := newTPM(tpm)
+	info, err := tpm1.GetCapVal(t)
+	if err != nil {
+		t.Close()
```
Comment (zaolin) [inline pkg/tss/tss.go:36]: use defer functionality

Comment (hugelgupf) [reply pkg/tss/tss.go:36]: No, t.Close actually seems appropriate here..?

**Comment by rminnich** [comment]
I would have tried to merge but there is a note about some other thing needing to be merged first? Looking forward to seeing this one come back but for now ... close it.

**Comment by ChriMarMe** [comment]
Force-pushed the branch, can't reopen. Dunno.

**Comment by rminnich** [comment]
I can'r reopen either. Damn. Can you just push to a new one and I'll review. Apologies.

**Comment by ChriMarMe** [comment]
Will do today or tomorrow. CI failed for change of dependencies, need to sort that out.

**Pull request closed without merging**